src/hooks/useLocalStorage.js

```diff
@@ -0,0 +1,39 @@
+import { useState } from "react";
+
+function useLocalStorage(key, initialValue) {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      // Get from local storage by key
+      const item = window.localStorage.getItem(key);
+      // Parse stored json or if none return initialValue
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      // If error also return initialValue
+      console.log(error);
```

Comment:
I'm not really a fan of the silent failure and this will be difficult to grep for.
```suggestion
      console.error('unable to parse local storage:', error);
```

Or have a shared error handler between these two try/catch blocks.